### PR TITLE
[Dubbo-2.7.5]Add arguments router to ConditionRouter

### PR DESCRIPTION
**Pull request by furaul**
Add arguments router logic to ConditionRouter.

## What is the purpose of the change

Add arguments router to ConditionRouter.
Compatible to current ConditionRouter logic.

## Brief changelog

Add arguments router to ConditionRouter

## Verifying this change

Add arguments router to ConditionRouter

## Examples
In some cases, we need to choose the right provider while the arguments of the method is setted.
The condition rule is like

> arguments[1].str!=b  => host = 1.1.1.1

While the str property of the second argument is equal to b, the providers whose host is equal to 1.1.1.1 are to be choosed。
We could use this feature to route requests according to business params. 
The feature is compatible to current ConditionRouter logic.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [x] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by chickenlj** [comment]
I think it would be more reasonable if we add a new independent Router dedicated to handling arguments.

**Comment by furaul** [comment]
> dedicated

Could you explain to me why we need a new independent Router for arguments?
你好，请帮忙解释下需要一个单独参数Router的原因呢？

**Comment by furaul** [comment]
> I think it would be more reasonable if we add a new independent Router dedicated to handling arguments.



> I think it would be more reasonable if we add a new independent Router dedicated to handling arguments.

Could you explain to me why we need a new independent Router for arguments?
你好，请帮忙解释下需要一个单独参数Router的原因呢？

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java [hunk 1]
```diff
@@ -1,301 +1,352 @@
...
-                        return true;
-                    }
-                }
-                return false;
-            }
-            return false;
-        }
-    }
-}
+/*
```
Comment (AlbumenJ) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java:1]: pls recover end-line format of this file

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java (hunk 1 above)
Comment (furaul) [reply dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/condition/ConditionRouter.java:1]: I have fix it in https://github.com/apache/dubbo/pull/7378, please refer to the new PR.

**Comment by furaul** [comment]
The format problem has been fixed in https://github.com/apache/dubbo/pull/7378, please refer to the new one. This PR would be closed later.

**Comment by AlbumenJ** [comment]
close this and continues discussion  in #7378

**Pull request closed without merging**